src/openms/source/ANALYSIS/OPENSWATH/PeakIntegrator.cpp

```diff
@@ -124,17 +124,25 @@ namespace OpenMS
 
     params.setValue("baseline_type", BASELINE_TYPE_BASETOBASE, "The baseline type to use in estimateBackground() based on the peak boundaries. A rectangular baseline shape is computed based either on the minimal intensity of the peak boundaries, the maximum intensity or the average intensity (base_to_base).");
     params.setValidStrings("baseline_type", ListUtils::create<String>("base_to_base,vertical_division,vertical_division_min,vertical_division_max"));
+
+    params.setValue("fit_EMG", "false", "Fit the chromatogram/spectrum to the EMG peak model.");
+    params.setValidStrings("fit_EMG", ListUtils::create<String>("false,true"));
   }
 
   void PeakIntegrator::updateMembers_()
   {
     integration_type_ = (String)param_.getValue("integration_type");
     baseline_type_ = (String)param_.getValue("baseline_type");
+    fit_EMG_ = param_.getValue("fit_EMG").toBool();
   }
 
   template <typename PeakContainerT>
-  PeakIntegrator::PeakArea PeakIntegrator::integratePeak_(const PeakContainerT& p, const double left, const double right) const
+  PeakIntegrator::PeakArea PeakIntegrator::integratePeak_(const PeakContainerT& pc, const double left, const double right) const
   {
+    PeakContainerT emg_pc;
+    if (fit_EMG_) emg_.fitEMGPeakModel(pc, emg_pc);
+    const PeakContainerT& p = fit_EMG_ ? emg_pc : pc;
```

Comment:
Looks good.  One thing that might be worth considering is to "extract" out the peak at the left and right and then call fit `EMGPeakModel` to avoid including other peaks.

Reply:
Nice catch.
It also made me notice something else:
The rest of the code in these three methods relies on `left` and `right`. This is ok, until we use the EMG fitting, because then at least in the case of cutoff peak one of the two boundaries becomes not reliable.
I could change the logic in a way that the `left` and `right` values are used only at the top of the methods.

Example:
```
if fit_emg is true
    spectrum = extract the part between left and right and pass it to the emg fitter
else
    spectrum = extract the part between left and right
integrate/estimateBG/computeMetrics on entire "spectrum" without relying on left and right
```

What do you think about it?

Reply:
I think that would work.

Reply:
I will look into a way to avoid the extraction step. Maybe by updating the `left` or `right` boundaries.